sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowFn.java

```diff
@@ -180,6 +180,13 @@ public boolean isNonMerging() {
   }
 
   /**
+   * Returns true if this {@link WindowFn} always assigns an element to exactly one window.
```

Comment:
`<p>If it varies per element, or cannot be determined, you should conservatively return false`